### DISCUSSION TITLE
Change setting on running self-hosted k8s deployment

### DISCUSSION THREAD

**Discussion by semanticfire** [plausible/analytics#4487] [Self-Hosted Support] [answered]
Hi,

I run Plausible CE 2.2.1 on k8s instance.
On deployment, I allowed registration to create the first account, but now I would like to make it invite only.
Is there a way to change these settings in a running instance?

Bart

**Comment by ruslandoga** [comment]
👋 @semanticfire 

Recreating the container with a new environment variable would make the new settings take effect. Note that the first account is always allowed to be created regardless of `DISABLE_REGISTRATION` setting.

With Docker it would've been something like this:

```shell
$ docker run --name plausible -e "DISABLE_REGISTRATION=true" ... ghcr.io/plausible/community-edition:v2.1.1
$ docker stop plausible && docker rm plausible && docker run -e "DISABLE_REGISTRATION=invite_only" ... ghcr.io/plausible/community-edition:v2.1.1
```

I'm not familiar enough with k8s to provide a better example :) But I guess k8s does something similar (recreating the container) whenever the ENV changes in the configuration files.

**Comment by semanticfire** [comment, marked as answer]
I've created the delpoyment based on the following yaml file:
https://gist.github.com/semanticfire/4831e643cdfe45f6ca7cca46271a4bb2

Here the settings environment settings are done in a secret on top of the file.

in my microk8s console, I've updated the secrets with proper values, and then scaled the deployment back to 0 and up to 1 to recreate the plausible app pod.

Problem solved!
